app/controllers/posts_controller.rb

```diff
@@ -27,6 +29,15 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @user = @post.author
+    @post.destroy
+    @user.posts_counter -= 1
+
+    redirect_to user_posts_path(@user) if @user.save
+  end
+
   private
```

Comment:
- [ ]  All attempts to delete the post is not successful and I would encourage you to take a double look at itit